src/EditorFeatures/CSharpTest/MisplacedUsings/MisplacedUsingsCodeFixProviderTests.cs

```diff
@@ -0,0 +1,1295 @@
+        /// </summary>
+        [Fact]
+        public Task WhenInsidePreferred_UsingsInCompilationUnit_UsingsMovedWithSystemPlacedFirst()
+        {
+            var testCode = @"using Microsoft.CodeAnalysis;
+using SystemAction = System.Action;
+using static System.Math;
+using System;
+
+using static System.String;
+using MyFunc = System.Func<int,bool>;
+
+using System.Collections.Generic;
+using System.Collections;
+
+namespace Foo
+{
+    public class Bar
+    {
+    }
+}
+";
+
+            var fixedTestCode = @"namespace Foo
+{
+    using System;
+    using System.Collections;
+    using System.Collections.Generic;
+    using Microsoft.CodeAnalysis;
+    using static System.Math;
+    using static System.String;
+    using MyFunc = System.Func<int,bool>;
+    using SystemAction = System.Action;
+
+    public class Bar
+    {
+    }
+}
+";
+
+            DiagnosticResult[] expected =
+            {
+                Diagnostic(MisplacedUsingsDiagnosticAnalyzer._insideDescriptor).WithLocation(1, 1),
+                Diagnostic(MisplacedUsingsDiagnosticAnalyzer._insideDescriptor).WithLocation(2, 1),
+                Diagnostic(MisplacedUsingsDiagnosticAnalyzer._insideDescriptor).WithLocation(3, 1),
+                Diagnostic(MisplacedUsingsDiagnosticAnalyzer._insideDescriptor).WithLocation(4, 1),
+                Diagnostic(MisplacedUsingsDiagnosticAnalyzer._insideDescriptor).WithLocation(6, 1),
+                Diagnostic(MisplacedUsingsDiagnosticAnalyzer._insideDescriptor).WithLocation(7, 1),
+                Diagnostic(MisplacedUsingsDiagnosticAnalyzer._insideDescriptor).WithLocation(9, 1),
+                Diagnostic(MisplacedUsingsDiagnosticAnalyzer._insideDescriptor).WithLocation(10, 1),
+            };
+            return VerifyCodeFixAsync(testCode, s_insidePreferenceOption, expected, fixedTestCode, placeSystemNamespaceFirst: true, separateImportDirectiveGroups: false);
+        }
+
+        /// <summary>
+        /// Verifies that the code fix will properly reorder using statements.
+        /// </summary>
+        /// <returns>A <see cref="Task"/> representing the asynchronous unit test.</returns>
+        [Fact]
+        public Task WhenInsidePreferred_UsingsInCompilationUnit_UsingsMovedWithSystemPlacedFirstInGroups()
+        {
+            var testCode = @"using Microsoft.CodeAnalysis;
+using SystemAction = System.Action;
+using static System.Math;
+using System;
+using static System.String;
+using MyFunc = System.Func<int,bool>;
+using System.Collections.Generic;
+using System.Collections;
+
+namespace Foo
+{
+    public class Bar
+    {
+    }
+}
+";
+
+            var fixedTestCode = @"namespace Foo
+{
+    using System;
+    using System.Collections;
+    using System.Collections.Generic;
+
+    using Microsoft.CodeAnalysis;
+
+    using static System.Math;
+    using static System.String;
+
+    using MyFunc = System.Func<int,bool>;
+    using SystemAction = System.Action;
```

Comment:
why are we doing more than we said we would do.  i'm not a big fan of 'move usings' now applying other things.  We already have 'organize' usings, which shoudl be doing the whole "move, sort, add blank lines".  But specific actions like "remove usings" "sort usings" or "move usings" should just do those single things.